src/js/components/presentational/InfiniteTable.jsx

```diff
@@ -6,14 +6,22 @@ const DEFAULT_DELAY = 1000
 class InfiniteTable extends Component {
   constructor (props) {
     super(props)
-    this.state = { scrollTop: 0 }
+    this.state = { scrollTop: 0,
+      scrollSpeed: DEFAULT_SPEED
+    }
   }
 
   componentDidMount() {
     setTimeout(() => {
       this.interval = setInterval(() => {
-          this.setState({ scrollTop: this.newScroll(this.state.scrollTop) })
-      }, 1000 / (this.props.speed || DEFAULT_SPEED))
+          this.setState({scrollTop: this.newScroll(this.state.scrollTop) })
+          if (this.state.scrollSpeed !== this.props.speed) {
+            console.log('scroll speed changed to ' + this.props.speed)
+            this.setState({scrollSpeed: this.props.speed})
+            clearInterval(this.interval)
+            this.componentDidMount()

```

Comment:
Why calling this function?

Reply:
The method is called to restart the interval that updates the scroll with a new timeout